Fix handling of ALGLIB dependency minimum required version

traversaro opened the pull request:
The minimum version needs to be specified in idyntree_handle_dependency via the MINIMUM_VERSION option

S-Dafarra commented:
Actually it's not just the minimum version. A few months ago I noticed that the compatibility was not preserved across versions. So, right now, only that specific version is compatible with the interface.

traversaro commented:
> Actually it's not just the minimum version. A few months ago I noticed that the compatibility was not preserved across versions. So, right now, only that specific version is compatible with the interface.

Ok, based on that I updated the version check. Given this peculiar characteristics of ALGLIB, I also added the `DO_NOT_SILENTLY_SEARCH` option, to only build the ALGLIB part if it is explicitly requested via `IDYNTREE_USES_ALGLIB` , otherwise if ALGLIB different from `3.14.0` was found, it would result in an configuration error.
